server/datastore/mysql/software_installers_test.go

```diff
@@ -3088,3 +3089,56 @@ func testBatchSetSoftwareInstallersActivateNextActivity(t *testing.T, ds *Datast
 	checkUpcomingActivities(t, ds, host2, host2Script.ExecutionID)
 	checkUpcomingActivities(t, ds, host3)
 }
+
+func testSaveInstallerUpdatesClearsFleetMaintainedAppID(t *testing.T, ds *Datastore) {
+	ctx := context.Background()
+	user := test.NewUser(t, ds, "Test User", "test@example.com", true)
+	tfr, err := fleet.NewTempFileReader(strings.NewReader("file contents"), t.TempDir)
+	require.NoError(t, err)
+
+	// Create an installer with a non-NULL fleet_maintained_app_id
+	installerID, _, err := ds.MatchOrCreateSoftwareInstaller(ctx, &fleet.UploadSoftwareInstallerPayload{
+		Title:           "testpkg",
+		Source:          "apps",
+		InstallScript:   "echo install",
+		PreInstallQuery: "SELECT 1",
+		UninstallScript: "echo uninstall",
+		InstallerFile:   tfr,
+		StorageID:       "storageid1",
+		Filename:        "test.pkg",
+		Version:         "1.0",
+		UserID:          user.ID,
+		ValidatedLabels: &fleet.LabelIdentsWithScope{},
+	})
+	require.NoError(t, err)
+
+	// Set fleet_maintained_app_id manually (simulate maintained app)
+	_, err = ds.writer(ctx).ExecContext(ctx, `UPDATE software_installers SET fleet_maintained_app_id = 42 WHERE id = ?`, installerID)
```

Comment:
Guessing this is breaking tests. FMA table needs to have something in it to satisfy the foreign key constraint.